docs/proposals/cloud-provider-refactoring.md

```diff
@@ -0,0 +1,121 @@
+## Refactor Cloud Provider out of Kubernetes Core
+
+As kubernetes has evolved tremendously, it has become difficult for different cloudproviders (currently 7) to make changes and iterate quickly. Moreover, the cloudproviders are constrained by the kubernetes build/release lifecycle. It has also become harder to add new cloud providers. This proposal aims to move towards a kubernetes code base where cloud providers specific code will move out of the core repository, and will be maintained by the cloud providers themselves.
```

Comment:
@luxas - even if it is in the "official" place, it will be maintained by the cloudprovider. I'm all for the official repository. I'll make it clearer in the above statement.